bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java

```diff
@@ -283,7 +283,7 @@ public Iterable<HistoricItem> query(FilterCriteria filter) {
         // set end to {@link Instant#MAX} instead of current timestamp to enable requesting future time ranges including
         // boundary values via REST API
         // see discussion in https://github.com/openhab/openhab-addons/pull/14238
-        long end = filter.getEndDate() == null ? Instant.MAX.getEpochSecond()
+        long end = filter.getEndDate() == null ? System.currentTimeMillis() / 1000
```

Comment:
This reintroduces the faulty logic fixed by #14238 as `end` can now end up being less than `start`, if `start` is in the future. An exception will no longer be thrown, only a warning logged. However, the logic is still not correct.

In the scenario where you are experiencing this, what is the concrete values for filter startDate and the current time? I'm wondering if the delta is only a few milliseconds, of is it's a actual future date.

Reply:
I'm aware, but until a better solution if found, with this change in place at least it works (compared to oom). I'm not at home so cannot dump concrete values, but assuming, based on [code](https://github.com/openhab/openhab-core/blob/8f9dafd1d68a92248cd39414145175952d61cd46/bundles/org.openhab.core.io.rest.core/src/main/java/org/openhab/core/io/rest/core/internal/persistence/PersistenceResource.java#L346-L351) delta might be different, depending on interval, i.e. imagine having per-month chart - based on selected month, `dateTimeEnd` might be (as set in api call):
```
endtime=2022-12-31T22:59:59.999Z
endtime=2022-11-30T22:59:59.999Z
...
```
depending on which month is being displayed/selected in chart view. And end time is used for start time
```
filterAfterEnd.setBeginDate(dateTimeEnd);
```
and no end time is set - resulting in different deltas, right?
```
delta = current timestamp - <2022-12-31T22:59:59.999Z>
```
or
```
delta = current timestamp - <2022-11-30T22:59:59.999Z>
```
...
As said, until a better solution is found, at least it works (renders chart) and probably logs a warning in some cases, but (IMHO) still better compared to not render charts at all due oom ...